sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -67,6 +67,8 @@ case class AdaptiveSparkPlanExec(
 
   @transient private val lock = new Object()
 
+  @transient private val logLevel = conf.adaptiveExecutionLogLevel
```

Comment:
nit: 
```
@transient private val log: String => Unit = conf.adaptiveExecutionLogLevel match {
  case "TRACE" => logTrace
  ...
}
```
to avoid checking the log level everytime.